docs/en/tutorials/model.md

```diff
@@ -1,3 +1,210 @@
+
+Makes sense but it's weird, isn't it? At this point you may be wondering:
+
+1. `MMResNet50` does not define `data_preprocessor`, but why it can still access `data_preprocessor` and move data to GPU?
+
+2. Why `BaseModel` does not move data by `data = data.to(device)`, but needs the `DataPreprocessor` to move data?
+
+The answer of the first question is that: `MMResNet50` inherit from `BaseModel`, and `super().__init__` will build a default `data_preprocessor` for it. The equivalent implementation of the default one like this:
+
+```python
+class BaseDataPreprocessor(nn.Module):
+    def forward(self, data, training=True):  # ignore the training parameter here
+        # suppose data given by CIFAR10 is a tuple. Actually
+        # BaseDataPreprocessor could move varies type of data
+        # to target device.
+        return tuple(_data.cuda() for _data in data)
+```
+
+`BaseDataPreprocessor` will move the data to the specified device.
+
+Before answering the second question, let's think about a few more questions
+
+1. Where should we perform normalization? [transform](../advanced_tutorials/data_transform.md) or `Model`?
+
+   It sounds reasonable to put it in transform to take advantage of Dataloader's multi-process acceleration, and in model to move it to GPU to use GPU resources to accelerate normalization. However, while we are debating whether CPU normalization is faster than GPU normalization, the time of data moving from CPU to GPU is much more longer than the former.
+
+   In fact, for less computationally intensive operations like normalization, it takes much less time than data transferring, which has a higher priority for being optimized. If I could move the data to the specified device while it is still in `uint8` and before it is normalized (the size of normalized `float` data is 4 times larger than that of unit8), it would reduce the bandwidth and greatly improve the efficiency of data transferring. This "lagged" normalization behavior is one of the main reasons why we designed the `DataPreprocessor`. The data preprocessor moves the data first and then normalizes it.
+
+2. How we implement the data augmentation like MixUp and Mosaic?
+
+   Although it seems that MixUp and Mosaic are just special data transformations that should be implemented in transform. However, consider that these two transformation involve **fusing multiple images into one**, it would be very difficult to implement them in transform since the current paradigm of transform is to do various enhancements on **one** image. It would be hard to read additional image from dataset because the dataset is not accessible in the transform. However, if we implement Mosaic or Mixup based on the `batch_data` sampled from Dataloader, everything becomes easy. We can access multiple images at the same time, and we can easily perform the image fusion operation.
```

Comment:
```suggestion
   Although it seems that MixUp and Mosaic are just special data transformations that should be implemented in transform. However, considering that these two transformations involve **fusing multiple images into one**, it would be very difficult to implement them in transform since the current paradigm of transform is to do various enhancements on **one** image. It would be hard to read additional images from dataset because the dataset is not accessible in the transform. However, if we implement Mosaic or Mixup based on the `batch_data` sampled from Dataloader, everything becomes easy. We can access multiple images at the same time, and we can easily perform the image fusion operation.
```